### PR TITLE
ec2_vpc_subnet retries

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
Improve ec2_vpc_subnet retry and exception handling
Add integration test suite

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ec2_vpc_subnet

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel 269672faf1) last updated 2017/10/18 17:05:01 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, Sep  5 2017, 08:53:59) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]

``

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py [hunk 1]
```diff
@@ -113,76 +113,87 @@ def get_subnet_info(subnet):
...
 
 
 def ensure_tags(conn, module, subnet, tags, add_only, check_mode):
-    try:
-        cur_tags = subnet['tags']
+    cur_tags = subnet['tags']
 
-        to_delete = dict((k, cur_tags[k]) for k in cur_tags if k not in tags)
-        if to_delete and not add_only and not check_mode:
+    to_delete = dict((k, cur_tags[k]) for k in cur_tags if k not in tags)
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py:158]: Can you use compare_aws_tags() for this?

**Review by s-hertel — APPROVED** [review]

File: lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py (hunk 1 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py:158]: @s-hertel this is mostly an indentation change.

You're quite right, but that would make this change 'bigger'

**Comment by willthames** [comment]
rebuild_merge

**Comment by willthames** [comment]
rebuild_merge

**Review by ryansb — APPROVED** [review]

**Pull request merged**